CMakeLists.txt

```diff
@@ -78,41 +78,54 @@ elseif (${CMINPACK_PRECISION} STREQUAL all)
   endif ()
 endif ()
 
+
+set(cminpack_lib_suffix "")
+
 if (BUILD_SHARED_LIBS)
   message (STATUS "Building shared libraries.")
-  if (NOT DEFINED CMAKE_DEBUG_POSTFIX)
-    set (CMAKE_DEBUG_POSTFIX _d)
-  endif ()
+  if ("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
+   set(cminpack_lib_suffix "_d")
```

Comment:
whats the rationale behind this and not relying on the global settings ?
sometimes you want to debug but not change the library name in order not to have to relink everything